framework/src/modules/keys/register_asset.ts

```diff
@@ -0,0 +1,134 @@
+			throw new Error('MandatoryKeys contains duplicate public keys.');
+		}
+
+		if (!objectUtils.bufferArrayUniqueItems(optionalKeys as Buffer[])) {
+			throw new Error('OptionalKeys contains duplicate public keys.');
+		}
+
+		// Check if key count is less than number of required signatures
+		if (mandatoryKeys.length + optionalKeys.length < numberOfSignatures) {
+			throw new Error(
+				'The numberOfSignatures is bigger than the count of Mandatory and Optional keys.',
+			);
+		}
+
+		// Check if key count is out of bounds
+		if (
+			mandatoryKeys.length + optionalKeys.length > MAX_KEYS_COUNT ||
+			mandatoryKeys.length + optionalKeys.length <= 0
+		) {
+			throw new Error('The count of Mandatory and Optional keys should be between 1 and 64.');
+		}
+
+		// The numberOfSignatures needs to be equal or bigger than number of mandatoryKeys
+		if (mandatoryKeys.length > numberOfSignatures) {
+			throw new Error(
+				'The numberOfSignatures needs to be equal or bigger than the number of Mandatory keys.',
+			);
+		}
+
+		// Check if keys are repeated between mandatory and optional key sets
+		const repeatedKeys = mandatoryKeys.filter(
+			value => optionalKeys.find(optional => optional.equals(value as Buffer)) !== undefined,
+		);
+		if (repeatedKeys.length > 0) {
+			throw new Error(
+				'Invalid combination of Mandatory and Optional keys. Repeated keys across Mandatory and Optional were found.',
+			);
+		}
+
+		// Check if the length of mandatory, optional and sender keys matches the length of signatures
+		if (mandatoryKeys.length + optionalKeys.length + 1 !== transaction.signatures.length) {
+			throw new Error(
+				'The number of mandatory, optional and sender keys should match the number of signatures',
+			);
+		}
+
+		// Check keys are sorted lexicographically
+		const sortedMandatoryKeys = [...mandatoryKeys].sort((a, b) => a.compare(b as Buffer));
+		const sortedOptionalKeys = [...optionalKeys].sort((a, b) => a.compare(b as Buffer));
+		for (let i = 0; i < sortedMandatoryKeys.length; i += 1) {
+			if (!mandatoryKeys[i].equals(sortedMandatoryKeys[i] as Buffer)) {
+				throw new Error('Mandatory keys should be sorted lexicographically.');
+			}
+		}
+
+		for (let i = 0; i < sortedOptionalKeys.length; i += 1) {
+			if (!optionalKeys[i].equals(sortedOptionalKeys[i] as Buffer)) {
+				throw new Error('Optional keys should be sorted lexicographically.');
+			}
+		}
+	}
+
+	public async applyAsset({ asset, stateStore, senderID }: ApplyAssetInput<Asset>): Promise<void> {
+		const sender = await stateStore.account.get<AccountKeys>(senderID);
+
+		// Check if multisignatures already exists on account
+		if (sender.keys.numberOfSignatures > 0) {
+			throw new Error('Register multisignature only allowed once per account.');
+		}
+
+		sender.keys = {
+			numberOfSignatures: asset.numberOfSignatures,
+			mandatoryKeys: asset.mandatoryKeys as Buffer[],
+			optionalKeys: asset.optionalKeys as Buffer[],
+		};
```

Comment:
```suggestion
		sender.keys = {
			...asset,
		};
```

Reply:
Prefer to keep like this as it was originally like this and maybe we don't want to store the reference to the object.